drivers/platform/stm32/stm32_gpio.h

```diff
@@ -63,11 +63,9 @@ struct stm32_gpio_init_param {
 struct stm32_gpio_desc {
 	/** Port */
 	GPIO_TypeDef *port;
-	/** Output mode */
-	uint32_t mode;
-	/** Speed grade */
-	uint32_t speed;
-};
+	/** GPIO configuration */
+	GPIO_InitTypeDef gpio_config;
+} stm32_gpio_desc;
```

Comment:
remove the `stm32_gpio_desc` tag from here, otherwise it generates this

```
[17:39:46] [LD] adxl355.o iio_adxl355.o iio_adxl355_trig.o no_os_gpio.o no_os_i2c.o no_os_irq.o no_os_spi.o stm32_delay.o stm32_gpio.o stm32_gpio_irq.o stm32_irq.o stm32_spi.o stm32_uart.o stm32_uart_stdio.o iio.o iio_app.o iio_trigger.o iiod.o common_data.o iio_trigger_example.o main.o parameters.o no_os_circular_buffer.o no_os_lf256fifo.o no_os_list.o no_os_util.o stm32f4xx_hal.o stm32f4xx_hal_cortex.o stm32f4xx_hal_dma.o stm32f4xx_hal_dma_ex.o stm32f4xx_hal_exti.o stm32f4xx_hal_flash.o stm32f4xx_hal_flash_ex.o stm32f4xx_hal_flash_ramfunc.o stm32f4xx_hal_gpio.o stm32f4xx_hal_pwr.o stm32f4xx_hal_pwr_ex.o stm32f4xx_hal_rcc.o stm32f4xx_hal_rcc_ex.o stm32f4xx_hal_spi.o stm32f4xx_hal_tim.o stm32f4xx_hal_tim_ex.o stm32f4xx_hal_uart.o generated_main.o stm32f4xx_hal_msp.o stm32f4xx_it.o stm32_gpio_irq_generated.o sysmem.o system_stm32f4xx.o
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_spi.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/iio/iio_app/iio_app.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/eval-adxl355-pmdz/src/common/common_data.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/eval-adxl355-pmdz/src/examples/iio_trigger_example/iio_trigger_example.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/eval-adxl355-pmdz/src/platform/stm32/main.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
/usr/lib/gcc/arm-none-eabi/11.2.0/../../../../arm-none-eabi/bin/ld: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/eval-adxl355-pmdz/src/platform/stm32/parameters.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: multiple definition of `stm32_gpio_desc'; /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/objs/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.o:/home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/app/Core/noos/drivers/platform/stm32/stm32_gpio.h:68: first defined here
collect2: error: ld returned 1 exit status
make[1]: *** [../../tools/scripts/generic.mk:338: /home/dari/Work/no-OS2/projects/eval-adxl355-pmdz/build/eval-adxl355-pmdz.elf] Error 1
make: *** [../../tools/scripts/generic.mk:300: all] Error 2
```